DOC Tidy up writing style of performance section (#766)
silverstripe/developer-docs: en/02_Developer_Guides/08_Performance/01_Object_Caching.md

## Overview

Object caching stores infrequently changing values to improve performance.
By default, the storage mechanism chooses the most performant adapter available

> (PHP opcache, APC, or filesystem). Other cache backends can be configured.

Reviewer: ```suggestion
By default, the storage mechanism chooses the most performant filesystem adapter available
(PHP opcache, or filesystem) but does not use in-memory cache unless you configure it. Other cache backends can be configured.
```
Oops, I missed that reference to APCu when removing that default for CMS 6.

Author: Updated

Object caching is particularly useful for:

- Storing the results of expensive database queries.
- Caching data retrieved from external APIs.

> - Reducing the load on frequently accessed resources.

Reviewer: The classic AI list of 3 :rofl:

I don't feel strongly about this suggestion so feel free to leave it if you prefer the list.

```suggestion
Object caching is useful for situations where fetching or generating data can take a long time, such as expensive database queries or API calls to external services.
```

Author: Updated

## Configuration

We use the [PSR-16](https://www.php-fig.org/psr/psr-16/) standard ("SimpleCache")

> for caching, through the [symfony/cache](https://symfony.com/doc/current/components/cache.html) library. Silverstripe provides a PSR-16 interface via the [Psr16Cache](https://github.com/symfony/cache/blob/6.1/Psr16Cache.php) class which wraps [PSR-6](https://www.php-fig.org/psr/psr-6/) usage in symfony/cache.

Reviewer: ```suggestion
for caching, through the [symfony/cache](https://symfony.com/doc/current/components/cache.html) library.

Silverstripe provides a PSR-16 interface via the [`Psr16Cache`](https://github.com/symfony/cache/blob/6.1/Psr16Cache.php) class which wraps `symfony/cache`'s [PSR-6](https://www.php-fig.org/psr/psr-6/) compliant adapters.
```

Author: Updated
